backend/src/util/VerifyFIN.ts

```diff
@@ -0,0 +1,34 @@
+const checkAlpha = ['X', 'W', 'U', 'T', 'R', 'Q', 'P', 'N', 'M', 'L', 'K'];
+const weightArray = [2, 7, 6, 5, 4, 3, 2];
+
+function validateFIN(fin: string): string {
+  const finUpper = fin.toUpperCase();
+  if (finUpper.length === 9) {
+    const sumArray = fin.slice(1, 8).split('').map(Number); // FIN String into array of numbers to multiply by weight
+
+    let sum = 0;
+    for (let i = 0; i < sumArray.length; i += 1) {
+      sum += sumArray[i] * weightArray[i];
+    } // Cross multiply with weightArray to get sum
+
+    if (finUpper[0] === 'G') {
+      sum += 4;
+    } // FIN that starts with G has their checkAlpha shifted by 4
+
+    if (checkAlpha[sum % 11] === finUpper.slice(-1)) {
+      return `${fin} is a valid FIN.`;
```

Comment:
Function might be easier to use if it returns a boolean if valid.

Reply:
Yeap I understand, tbh many of these are still mental placeholders because I'm still not sure how the current stack works. The code is still very raw but the logic is there, feel free to change around with them, appreciate the inputs!